generators/chipyard/src/main/scala/harness/HarnessBinders.scala

```diff
@@ -6,359 +6,284 @@ import chisel3.experimental.{Analog, BaseModule, DataMirror, Direction}
-    val p: Parameters = chipyard.iobinders.GetSystemParameters(system)
-    (ports zip system.memAXI4Node.edges.in).map { case (port, edge) =>
-      val mem = LazyModule(new SimAXIMem(edge, size=p(ExtMem).get.master.size)(p))
-      Module(mem.module).suggestName("mem")
-      mem.io_axi4.head <> port.bits
-    }
+class WithSimAXIMem extends HarnessBinder({
+  case (th: HasHarnessInstantiators, port: AXI4MemPort) => {
+    val mem = LazyModule(new SimAXIMem(port.edge, size=port.params.master.size)(Parameters.empty))
+    withClock(port.io.clock) { Module(mem.module) }
+    mem.io_axi4.head <> port.io
   }
 })
 
-class WithBlackBoxSimMem(additionalLatency: Int = 0) extends OverrideHarnessBinder({
-  (system: CanHaveMasterAXI4MemPort, th: HasHarnessInstantiators, ports: Seq[ClockedAndResetIO[AXI4Bundle]]) => {
-    val p: Parameters = chipyard.iobinders.GetSystemParameters(system)
-    (ports zip system.memAXI4Node.edges.in).map { case (port, edge) =>
-      // TODO FIX: This currently makes each SimDRAM contain the entire memory space
-      val memSize = p(ExtMem).get.master.size
-      val memBase = p(ExtMem).get.master.base
-      val lineSize = p(CacheBlockBytes)
-      val clockFreq = p(MemoryBusKey).dtsFrequency.get
-      val mem = Module(new SimDRAM(memSize, lineSize, clockFreq, memBase, edge.bundle)).suggestName("simdram")
-      mem.io.axi <> port.bits
-      // Bug in Chisel implementation. See https://github.com/chipsalliance/chisel3/pull/1781
-      def Decoupled[T <: Data](irr: IrrevocableIO[T]): DecoupledIO[T] = {
-        require(DataMirror.directionOf(irr.bits) == Direction.Output, "Only safe to cast produced Irrevocable bits to Decoupled.")
-        val d = Wire(new DecoupledIO(chiselTypeOf(irr.bits)))
-        d.bits := irr.bits
-        d.valid := irr.valid
-        irr.ready := d.ready
-        d
-      }
-      if (additionalLatency > 0) {
-        withClockAndReset (port.clock, port.reset) {
-          mem.io.axi.aw <> (0 until additionalLatency).foldLeft(Decoupled(port.bits.aw))((t, _) => Queue(t, 1, pipe=true))
-          mem.io.axi.w  <> (0 until additionalLatency).foldLeft(Decoupled(port.bits.w ))((t, _) => Queue(t, 1, pipe=true))
-          port.bits.b   <> (0 until additionalLatency).foldLeft(Decoupled(mem.io.axi.b))((t, _) => Queue(t, 1, pipe=true))
-          mem.io.axi.ar <> (0 until additionalLatency).foldLeft(Decoupled(port.bits.ar))((t, _) => Queue(t, 1, pipe=true))
-          port.bits.r   <> (0 until additionalLatency).foldLeft(Decoupled(mem.io.axi.r))((t, _) => Queue(t, 1, pipe=true))
-        }
+class WithBlackBoxSimMem(additionalLatency: Int = 0) extends HarnessBinder({
+  case (th: HasHarnessInstantiators, port: AXI4MemPort) => {
+    // TODO FIX: This currently makes each SimDRAM contain the entire memory space
+    val memSize = port.params.master.size
+    val memBase = port.params.master.base
+    val lineSize = 64 // cache block size
```

Comment:
LineSize isn't parameterized anymore?

Reply:
line_size doesn't actually matter for DRAM. We shouldn't even be passing it back to DRAMSim in the first place (and our DRAMSim driver doesn't use the line_size parameter for any calculations).  It also requires line_size=64.

Future TODO is to remove the line_size argument to DRAMSim, so we can get rid of this here as well too.